addons/website_sale_slides/static/src/js/slides_course_quiz.js

```diff
@@ -20,6 +20,28 @@ sAnimations.registry.websiteSlidesQuizNoFullscreen.include({
 Quiz.include({
     xmlDependencies: (Quiz.prototype.xmlDependencies || []).concat(
         ["/website_sale_slides/static/src/xml/website_sale_slides_quiz.xml"]
-    )
+    ),
+    events: _.extend({}, Quiz.prototype.events || {}, {
+        'click .o_wslides_js_join_course_sale': '_onClickJoinSale',
+    }),
+    /**
+     * @private
+     */
+    _onClickJoinSale: function (event) {
+        event.preventDefault();
+        var url = _.str.sprintf('/shop/cart/update?product_id=%s&amp;express=1', (this.channel.productId));
```

Comment:
This part is common: `_.str.sprintf('/shop/cart/update?product_id=%s&amp;express=1', (this.channel.productId));`

You could put it in a variable like `baseShopUrl` and pass it to your `_getLoginRedirectUrlSale` method to avoid duplicating it.